huaweicloud/services/acceptance/rds/resource_huaweicloud_rds_pg_plugin_test.go

```diff
@@ -0,0 +1,106 @@
+package rds
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getPgPluginResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	getPgPluginClient, err := cfg.NewServiceClient("rds", acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating RDS client: %s", err)
+	}
+
+	listPgPluginHttpUrl := "v3/{project_id}/instances/{instance_id}/extensions?database_name={database_name}"
+	listPgPluginPath := getPgPluginClient.Endpoint + listPgPluginHttpUrl
+	listPgPluginPath = strings.ReplaceAll(listPgPluginPath, "{project_id}", getPgPluginClient.ProjectID)
+	listPgPluginPath = strings.ReplaceAll(listPgPluginPath, "{instance_id}", state.Primary.Attributes["instance_id"])
+	listPgPluginPath = strings.ReplaceAll(listPgPluginPath, "{database_name}", state.Primary.Attributes["database_name"])
+
+	resp, err := getPgPluginClient.Request("GET", listPgPluginPath, &golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving RDS PostgreSQL : %s", err)
+	}
+
+	body, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving  RDS PostgreSQL : %s", err)
+	}
+
+	name := state.Primary.Attributes["name"]
+	plugin := utils.PathSearch(fmt.Sprintf("extensions[?name=='%s']|[?created]|[0]", name), body, nil)
+
+	if plugin == nil {
+		return nil, fmt.Errorf("error, no RDS PostgreSQL  matching %s was found", name)
```

Comment:
ditto, and update the error message

```suggestion
		return nil, fmt.Errorf("no RDS PostgreSQL plugin matching %s was found", name)
```